torch/_dynamo/utils.py

```diff
@@ -2336,6 +2336,12 @@ def is_int_specialization_case(value, source):
             source.guard_source().is_specialized_nn_module()
             and not config.allow_unspec_int_on_nn_module
         )
+        # integers coming from FSDP modules are considered static. This is
```

Comment:
so the current state today is that integers coming into FSDP modules are *only* considered static if `_unsafe_skip_fsdp_module_guards = True`, I think?

Given that normal OSS code today actually now defaults these int inputs defaulting to dynamic, do you think we should try to keep that current state, and just give a (similar) config to opt into this FSDP behavior?